client/src/document/ingredients/browser-compatibility-table/index.tsx

```diff
@@ -1,99 +1,57 @@
 import React from "react";
-import { Platforms } from "./platforms";
-import { Browsers } from "./browsers";
-import { Rows } from "./rows";
-import { Legend } from "./legend";
-import "./bcd.scss";
+import type bcd from "mdn-browser-compat-data/types";
+import { AccordionProvider } from "./accordion";
+import { BrowserInfoContext } from "./browser-info";
 import { BrowserCompatibilityErrorBoundary } from "./error-boundary";
+import { FeatureRow } from "./feature-row";
+import { PLATFORM_BROWSERS, Headers } from "./headers";
+import { Legend } from "./legend";
+import { listFeatures } from "./utils";
 
-const BROWSERS = {
-  desktop: ["chrome", "edge", "firefox", "ie", "opera", "safari"],
-  mobile: [
-    "webview_android",
-    "chrome_android",
-    "firefox_android",
-    "opera_android",
-    "safari_ios",
-    "samsunginternet_android",
-  ],
-  server: ["nodejs"],
-  "webextensions-desktop": ["chrome", "edge", "firefox", "opera"],
-  "webextensions-mobile": ["firefox_android"],
-};
-
-class BrowserCompatibilityTableContent extends React.Component<any, any> {
-  state = {
-    currentNoteId: null,
-    hasDeprecation: false,
-    hasExperimental: false,
-    hasNonStandard: false,
-    hasAlternative: false,
-    hasFlag: false,
-    hasPrefix: false,
-    hasNotes: false,
-    legendSet: false,
-  };
+import "./bcd.scss";
 
-  gatherPlatformsAndBrowsers(category) {
-    let platforms = ["desktop", "mobile"];
-    let displayBrowsers = [...BROWSERS["desktop"], ...BROWSERS["mobile"]];
-    if (category === "javascript") {
-      displayBrowsers.push(...BROWSERS["server"]);
-      platforms.push("server");
-    }
-    if (category === "webextensions") {
-      displayBrowsers = [
-        ...BROWSERS["webextensions-desktop"],
-        ...BROWSERS["webextensions-mobile"],
-      ];
-      platforms = ["webextensions-desktop", "webextensions-mobile"];
-    }
-    return [platforms, displayBrowsers];
+function gatherPlatformsAndBrowsers(category): [string[], bcd.BrowserNames[]] {
+  let platforms = ["desktop", "mobile"];
+  if (category === "javascript") {
+    platforms.push("server");
+  } else if (category === "webextensions") {
+    platforms = ["webextensions-desktop", "webextensions-mobile"];
   }
+  return [
+    platforms,
+    platforms.map((platform) => PLATFORM_BROWSERS[platform] || []).flat(),
+  ];
+}
 
-  onNotesClick = (noteId) => {
-    this.setState({
-      currentNoteId: noteId === this.state.currentNoteId ? null : noteId,
-    });
-  };
+export function BrowserCompatibilityTable({
+  id,
+  title,
+  query,
+  data,
+  browsers: browserInfo,
+}: {
+  id: string;
+  title: string;
+  query: string;
+  data: bcd.Identifier;
```

Comment:
What is a `bcd.Identifier`??
Judging from the check on line 39, I'm guessing this is a form of `object`. No?

Reply:
Here's an example of an Identifier object:
https://github.com/mdn/browser-compat-data/blob/12d83c4e9396093a9f24f746508356d01b6c90e4/api/Document.json#L99-L146

I also find the naming a bit strange.